### ISSUE TITLE
Committing new post produces error with Github

### ISSUE THREAD

**Issue by cmsS62** [prose/prose#1065]
Creating new post in prose produces GitHub error: See attached screenshot.

<img width="131" alt="screenshot 2018-02-28 17 19 15" src="https://user-images.githubusercontent.com/36015893/36817358-dfc294d2-1cae-11e8-84a9-041347dfd9f2.png">

**Comment by dereklieu** [comment]
Cross-origin issue related to ssl, going to push a fix for this and any other hard-coded http calls.

**Comment by dereklieu** [comment]
Actually this isn't a mixed-content error and I'm not sure what's going on, but I've put in a support ticket to Github. As far as I can tell it's a result of switching to https, as this was working yesterday and works from localhost.

Now I'm just getting this:

<img width="709" alt="screen shot 2018-02-28 at 7 08 33 pm" src="https://user-images.githubusercontent.com/1590802/36820218-cca46dba-1cba-11e8-9d29-bd958799cedd.png">

**Comment by cmsS62** [comment]
Thanks @dereklieu for looking into this! Can't see the newly resulting message (screenshot) though; it says it's forbidden content!?

**Comment by mattdock** [comment]
Hey, not sure if it is related at all but Prose was doing this for me as well and for some dumb reason I thought "Maybe something is wrong with the auth" so I revoked Prose's access then went back to re-authorize Prose and Prose just stayed a blank page.

Checking the console shows a 401 error - except I logged out before hand.

On that note - logging out doesn't log out (which may be the issue too).  It removed the logout button but I was still accessing my stuff - and when I tried to go to prose.io and reset my browser cache it was still showing my account's content(my friend wanted to test this on his Pages site.)

... At his point I should just make another thread.

@cmsS62 if you've got tips, i'd love to hear them.

**Comment by dereklieu** [comment]
@mattdock thanks for the tip. Logging out will remove the current oauth token from your local storage. You should still be able to view public repos while not logged in, though you won't be able to commit anything (this itself may be a UI problem though).

It seems this is an issue with the https version of Prose. http://prose.io is still working as expected, and we've temporarily stopped forcing https until this is fixed.

**Comment by cmsS62** [comment]
@mattdock No further insights, just as @dereklieu mentioned: " http://prose.io is still working as expected".

**Comment by mattdock** [comment]
@dereklieu @cmsS62 

<img width="421" alt="screen shot 2018-03-01 at 2 40 21 pm" src="https://user-images.githubusercontent.com/36573170/36873852-a68a5c8a-1d5e-11e8-8af6-802f4c3b2c12.png">


The github status is good - but cannot access Prose.
I need to re-authenticate (as mentioned in the prior post - i revoked Proses permission while trying to edit again) but this is all I am seeing.

Probably needs to prompt to authorize on a 401 instead of showing this.

Unless there is something I need to do in my profile settings?

**Comment by dereklieu** [comment]
@mattdock can you log out, clear your cookies for prose.io, and try logging in again?

**Comment by mattdock** [comment]
ugh - idk what was happening - cleared my cache for just prose.io and it was still happening so i cleared it for github and was still happening.

Decided to clear everything and it's finally letting me back in - but now i'm back to the same issue as the OP and have to remember passwords on all my sites that I haven't had to enter in more than a year LOL

Sorry for the trouble everyone

**Comment by zamber** [comment]
@mattdock use a password manager ;).

I was able to reproduce the issue with HTTPS (that's why I searched for this issue).
I'm using the HTTPS version (HTTPS Everywhere ftw).

Fonts, CSS and JS is not served over HTTPS. This may pose issues with doing xhr to api.github.com (it doesn't now, just fonts are not loaded).
 
Also I see 404's for https://prose.io/api.github.com/repos/zamber/zamber.github.io/contents/_config.yml?ref=master on the HTTPS version where on HTTP this request is done directly to api.github.com.

Seems like that's the root cause. Yep, confirmed by switching to HTTP where the request is not done with https://prose.io/api.github.com but directly to api.github.com.

I'm using Firefox 58.0.2 (64-bit) on Windows 10.

Let's see if I can find where this happens in the code...
